subprojects/build-cache/build-cache.gradle.kts

```diff
@@ -25,8 +25,10 @@ plugins {
 dependencies {
     api(project(":baseServices"))
     api(library("inject"))
+    api(project(":coreApi"))
 
     implementation(project(":messaging"))
+    implementation(project(":modelCore"))
```

Comment:
Where is either of these needed? I don't see a reference to `model-core` or `core-api` in here:

![image](https://user-images.githubusercontent.com/495366/50886616-32182500-13f1-11e9-8a22-7ae0feb8544e.png)

Reply:
Since `Instantiator` stayed in `baseServices`, we don't need a dependency on `modelCore` - we should maybe switch to `InstantiatorFactory` though, as this is the recommended way to obtain an instantiator.

There is a dependency on `coreApi` though, namely the `BuildCache` public types.
![image](https://user-images.githubusercontent.com/423186/50889159-8e7e4300-13f7-11e9-8dc3-630b852f1db0.png)

What do you think? Shall we switch to using `InstatiatorFactory.injectAndDecorate`?

Reply:
I'd not make that change now. Where is `InstantiationFactory` recommended?

Reply:
Here: https://github.com/gradle/gradle/blob/d2f7a13ff1cc66b10fab725d84988d9203856bf6/subprojects/base-services/src/main/java/org/gradle/internal/reflect/Instantiator.java#L35

Reply:
Let's create a card for adopting `InstantiatorFactory`.

Damn, I didn't see the dependency on `core-api` because IntelliJ had compile errors due to changes to build configuration. OK, so I guess we need to add `core-api`.

Reply:
I removed the `modelCore` dependency. I don't think it is worth to create the card to adopt `InstantiatorFactory` since this is not restricted to `BuildCacheConfiguration`.